xarray/core/missing.py

```diff
@@ -549,8 +549,8 @@ def _localize(var, indexes_coords):
     indexes = {}
     for dim, [x, new_x] in indexes_coords.items():
         index = x.to_index()
-        imin = index.get_loc(np.min(new_x.values), method="nearest")
-        imax = index.get_loc(np.max(new_x.values), method="nearest")
+        imin = index.get_loc(np.nanmin(new_x.values), method="nearest")
+        imax = index.get_loc(np.nanmax(new_x.values), method="nearest")
```

Comment:
At first I was a little uneasy about your suggestion to require a newer version of NumPy, but after reading about the history of this issue a little more, I think something along those lines may actually be the simplest approach (perhaps you were ahead of me on this).  

It turns out that `np.min` and `np.max` behave like `np.nanmin` and `np.nanmax` for time-like types in versions of NumPy prior to 1.18.1 (see https://github.com/numpy/numpy/pull/14841#issuecomment-554359720), so we can continue to use those for time-like types and earlier versions of NumPy and still obtain the desired behavior.  I think something like the following would work:
```suggestion
        if LooseVersion(np.__version__) < LooseVersion("1.18.1") and new_x.dtype.kind in "mM":
            # In older versions of NumPy min and max (incorrectly) ignore NaT values
            # for time-like types -- that is they behave like nanmin and nanmax
            # should -- but nanmin and nanmax raise an error. 
            imin = index.get_loc(np.min(new_x.values), method="nearest")
            imax = index.get_loc(np.max(new_x.values), method="nearest")
        else:
            imin = index.get_loc(np.nanmin(new_x.values), method="nearest")
            imax = index.get_loc(np.nanmax(new_x.values), method="nearest")
```